src/utils/workspace-context-utils.ts

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import { generatePath, useNavigate } from 'react-router-dom';
+import {
+  getActiveWorkspace,
+  k8sListResourceItems,
+  K8sResourceCommon,
+  setActiveWorkspace,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import { WorkspaceModel } from '../models/workspace';
+import { useSetNamespace } from './namespace-context-utils';
+
+const workspacePathMatcher = new RegExp(/workspaces\/([^/]+)\/applications/);
+
+export type WorkspaceContextData = {
+  workspace: string;
+  workspacesLoaded: boolean;
+  workspaces: K8sResourceCommon[];
```

Comment:
Perhaps it would be helpful to define a type for workspace CR? Or maybe it's not needed at this point?

Reply:
Yeah, I think its a good idea to define workspace model.

Reply:
Added workspace type.